File: services/notification-service/src/main/java/com/mealmap/notificationservice/core/dto/notification/NotificationDto.java
package com.mealmap.notificationservice.core.dto.notification;

import com.mealmap.notificationservice.doc.enums.Channel;
import com.mealmap.notificationservice.doc.enums.NotificationStatus;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import org.bson.types.ObjectId;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@Schema(description = "Информация об уведомлении")
public class NotificationDto {
    private ObjectId id;

    private String userId;

    private Channel channel;

    private NotificationStatus status;

    private String subject;

    private String message;

    private LocalDateTime sentAt;
}

File: frontend/src/api/notification/dto/NotificationDto.ts
import { Channel } from "../enums/Channel";
import { NotificationStatus } from "../enums/NotificationStatus";

export interface NotificationDto {
  id: string; // ObjectId as a string
  userId: string;
  channel: Channel;
  status: NotificationStatus;
  subject: string;
  message: string;
  sentAt: string; // ISO date-time string
}
